TESTS/mbed_hal/lp_ticker/main.cpp

```diff
@@ -104,6 +104,26 @@ void lp_ticker_glitch_test()
     }
 }
 
+void lp_ticker_consecutive_set_interrupt(void)
+{
+    intFlag = 0;
+
+    const ticker_info_t* p_ticker_info = lp_ticker_get_info();
+
+    set_lp_ticker_irq_handler(ticker_event_handler_stub);
+
+    lp_ticker_init();
+
+    const uint32_t tick_count = lp_ticker_read();
+
+    lp_ticker_set_interrupt(0);
```

Comment:
Please note that setting interrupt to some constant value like 0 is hazardous. Init function does not resets the ticker counter, so in some specific case if we are just about to overflow while setting the first interrupt it may be fired immediately (before the interrupt is rescheduled). And after wait `intFlag` might be equal to 2.

Additionally I think we have such test case already: https://github.com/ARMmbed/mbed-os/blob/c9ecc9b2db33f419e59140bb2309095b325fe6b6/TESTS/mbed_hal/common_tickers/main.cpp#L227

Reply:
How about
```
const uint32_t tick_count = lp_ticker_read();

 lp_ticker_set_interrupt(tick_count - 1);
```
?

Reply:
This would be fine, but do we really need this? What is the difference between this case and `ticker_repeat_reschedule_test` test case?

We have this tested for both tickers in `tests-mbed_hal-common_tickers`.

Reply:
Ah actually you may be right. The purpose of this test was to catch flawed implementations. For some HW you need to wait for ACK from the bus that the value was committed, it may take up to couple of tick of 32k clock. So if you don't check whether the value was committed the subsequent calls will be ignored. But test you mentioned should cover it.

Reply:
I do not know about the bug which has not been caught by the `ticker_repeat_reschedule_test`. I assume that it is about successive `set_interrupt()` calls, so it is possible that when we execute the following code:
```
const uint32_t tick_count = lp_ticker_read();
lp_ticker_set_interrupt(tick_count - 1);
lp_ticker_set_interrupt(tick_count + TICKER_INT_VAL);
```
Then the second request will be ignored and interrupt will not be fired at `tick_count + TICKER_INT_VAL` point in time?

I can not see the difference between proposed solution and current one where we have:
```
const uint32_t tick_count = intf->read();

intf->set_interrupt(tick_count + TICKER_INT_VAL);
intf->set_interrupt(tick_count + (2 * TICKER_INT_VAL));
intf->set_interrupt(tick_count + (3 * TICKER_INT_VAL));
```

And we expect that interrupt will be fired at `tick_count + (3 * TICKER_INT_VAL)`.

On which platform we have such problem? 
Does `ticker_repeat_reschedule_test` test case always passes on this platform if the test is run several times? 
Does `lp_ticker_consecutive_set_interrupt` test case always fails on this platform if the test is run several times?